import "library/cpp/getoptpb/proto/confoption.proto";

import "ads/bsyeti/big_rt/lib/consuming_system/config/config.proto";
import "ads/bsyeti/big_rt/lib/processing/shard_processor/stateless/config/stateless_shard_processor_config.proto";
import "ads/bsyeti/big_rt/lib/supplier/config/supplier_config.proto";

import "ads/bsyeti/libs/profiling/solomon/proto/config.proto";
import "ads/bsyeti/libs/tvm_manager/proto/config.proto";
import "ads/bsyeti/libs/ytex/http/proto/config.proto";
import "ads/bsyeti/libs/ytex/logging/proto/config.proto";

package NAlice.NWonderlogs;

message TMegamindResharderConfig {
    message TShardingConfig {
        message TSplitter {
            optional string LogfellerSplitter = 1 [default = "\n"];
        }

        optional TSplitter Splitter = 1;
        optional string ShardingAlgorithm = 2 [default = "by_uuid"];
        optional uint64 SkipUsersPercent = 3;
        optional uint64 ShardsCount = 4;
        optional string OutputQueue = 5;
        optional string OutputCodec = 6;
    }

    optional TShardingConfig ShardingConfig = 1;

    optional NYTEx.NLogging.TConfig Logs = 2;
    optional NYTEx.NHttp.TServerConfig HttpServer = 3;
    optional NBSYeti.NProfiling.TExporterConfig SolomonExporter = 4;

    repeated NBigRT.TSupplierConfig Suppliers = 5;
    optional NBigRT.TConsumingSystemConfig ConsumingSystem = 6;
    optional NBigRT.TStatelessShardProcessorConfig StatelessShardProcessorConfig = 7;

    optional uint64 MaxInflightBytes = 8;

    optional uint32 Port = 9;

    optional NBSYeti.TTvmGlobalConfig TvmConfig = 10;
}
